src/test/java/com/gachigang/ontherun/service/ProductServiceTest.java

```diff
@@ -63,41 +69,91 @@ public void testGetAllByCategory() {
+                .id(1L)
+                .build();
 
-        Category category = new Category();
-        Department department = new Department();
+        Department department = Department.builder()
+                .id(2L)
+                .build();
 
-        Mockito.when(categoryRepository.findById(1L)).thenReturn(Optional.of(category));
-        Mockito.when(departmentRepository.findById(1L)).thenReturn(Optional.of(department));
-        Mockito.when(productRepository.save(product)).thenReturn(product);
+        Mockito.when(categoryService.findById(1L)).thenReturn(category);
+        Mockito.when(departmentService.findByID(2L)).thenReturn(department);
+        Mockito.when(productRepository.save(Mockito.any(Product.class))).thenReturn(product);
 
-        Product result = productService.createProduct(product);
+        Product createdProduct = productService.createProduct(product);
 
-        assertNotNull(result);
-        assertEquals(category, result.getCategory());
-        assertEquals(department, result.getDepartment());
-        assertTrue(result.getActive());
+        Assertions.assertNotNull(createdProduct);
+        Assertions.assertEquals(category, createdProduct.getCategory());
+        Assertions.assertEquals(department, createdProduct.getDepartment());
+
+        Mockito.verify(categoryService, Mockito.times(1)).findById(1L);
+        Mockito.verify(departmentService, Mockito.times(1)).findByID(2L);
+        Mockito.verify(productRepository, Mockito.times(1)).save(product);
     }
 
 
     @Test
-    public void testHideProduct() {
-        Long productId = 1L;
-        Product product = new Product();
-        product.setActive(true);
+    public void testUpdateProduct() {
+
+        Product existingProduct = Product.builder()
+                .id(1L)
+                .categoryId(1L)
+                .departmentId(2L)
+                .build();
+
+        Product updatedProduct = Product.builder()
+                .id(1L)
+                .categoryId(3L)
+                .departmentId(4L)
+                .build();
+
+        Category category = Category.builder()
+                .id(3L)
+                .build();
+
+        Department department = Department.builder()
+                .id(4L)
+                .build();
+
+        Mockito.when(productRepository.existsProductById(1L)).thenReturn(true);
+        Mockito.when(categoryService.findById(3L)).thenReturn(category);
+        Mockito.when(departmentService.findByID(4L)).thenReturn(department);
+        Mockito.when(productRepository.save(Mockito.any(Product.class))).thenReturn(updatedProduct);
+
+        Product updated = productService.updateProduct(updatedProduct);
+
+        Assertions.assertNotNull(updated);
+        Assertions.assertEquals(category, updated.getCategory());
+        Assertions.assertEquals(department, updated.getDepartment());
+
+        Mockito.verify(productRepository, Mockito.times(1)).existsProductById(1L);
```

Comment:
We need to write asserts OR verifies -- So, since you have already used asserts, no need to write verifies.